.github/workflows/tests.yml

```diff
@@ -110,90 +140,127 @@ jobs:
 
     - name: Install Ingress
       run: |
-        kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v${{ matrix.ingress }}/deploy/static/provider/baremetal/deploy.yaml
-        sleep 5
-        time kubectl -n ingress-nginx wait --for=condition=Available deployment --all --timeout 300s
-
-    - uses: actions/checkout@v2
-    - name: go tests
-      run: |
-        make go-test
-
-    - name: fetch logs
-      if: ${{ failure() }}
-      run: |
-        mkdir  -p /tmp/logs
-        kind export logs --name harbor /tmp/logs
-
-    - uses: actions/upload-artifact@v2
-      if: ${{ failure() }}
-      with:
-        name: kind
-        path: /tmp/logs
+        kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v${{ matrix.ingress }}/deploy/static/provider/kind/deploy.yaml
+        time kubectl wait --namespace ingress-nginx --for=condition=ready pod --selector=app.kubernetes.io/component=controller --timeout=300s
 
     - name: build harbor-operator
       run: |
-        make docker-build IMG=${dockerImage}
+        make manifests docker-build IMG=${dockerImage}
         kind load docker-image ${dockerImage} --name harbor
 
     - name: install harbor-operator
       run: |
-        set -ex
-        make helm-install NAMESPACE="${operatorNamespace}" IMG=${dockerImage}
-        kubectl -n "${operatorNamespace}" wait --for=condition=Available deployment --all --timeout 300s
+        cd manifests/cluster
+        kustomize edit add secret github-token --disableNameSuffixHash --from-literal=GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
+        kustomize edit add patch --path patch/github-token.yaml
+        kustomize edit set image goharbor/harbor-operator=${dockerImage}
+        kustomize build | kubectl apply -f -
+
+        # make helm-install IMG=${dockerImage}
+
+        if ! time kubectl -n ${operatorNamespace} wait --for=condition=Available deployment --all --timeout 300s; then
+          kubectl get all -n ${operatorNamespace}
+          exit 1
+        fi
```

Comment:
I suggest to keep the *helm install* way:
- IMO it is closer to the reality than `kustomize build | kubectl apply -f -`
- Multiple `kustomize edit ...` before running test looks like default values do not work.